### ISSUE TITLE
length(Range{BigInt}) returns BigInt. Issue for two argument vectorized functions.

### ISSUE THREAD

**Issue by andreasnoack** [JuliaLang/julia#15927]
Much of Base assumes that `size` and `length` return `Int`. It's the most frequent value

``` jl
julia> countmap(Base.return_types(length, Tuple{AbstractVector})) |> t -> [map(string, keys(t)) map(string, values(t))]
5x2 Array{ASCIIString,2}:
 "Union{BigInt,Int64}"  "1" 
 "Any"                  "5" 
 "Integer"              "2" 
 "Int64"                "10"
 "UInt64"               "1" 
```

(dict printing is still not working) but other values are possible and they can occationally give errors. I just ran into

``` jl
julia> lbeta(big(1:4), 1:4)
ERROR: MethodError: no method matching promote_shape(::Tuple{BigInt}, ::Tuple{Int64})
Closest candidates are:
  promote_shape(::Tuple{Int64}, ::Tuple{Int64})
  promote_shape(::Tuple{Int64,Int64}, ::Tuple{Int64})
  promote_shape(::Tuple{Vararg{Int64}}, ::Tuple{Vararg{Int64}})
 in lbeta(::UnitRange{BigInt}, ::UnitRange{Int64}) at ./operators.jl:499
 in eval(::Module, ::Any) at ./boot.jl:237
```

which would also be a problem for a `Range{UInt64}`. Another example where we assume `Int` values is in 

``` jl
julia> similar(randn(10), size(big(1:3)),)
ERROR: MethodError: no method matching Array{T,N}(::Tuple{BigInt}, ::Int64)
Closest candidates are:
  (::Type{BoundsError})(::ANY, ::ANY)
  (::Type{TypeError})(::Any, ::Any, ::Any, ::Any)
  (::Type{TypeConstructor})(::ANY, ::ANY)
  ...
 in similar(::Array{Float64,1}, ::Tuple{BigInt}) at ./array.jl:125
 in eval(::Module, ::Any) at ./boot.jl:237
```

I guess the fix is to allow all kinds of integer types as input since it probably doesn't make sense to restrict e.g. `length(UnitRange)` to return an `Int`.

**Comment by H-225** [comment]
> (dict printing is still not working)

See PR https://github.com/JuliaLang/julia/pull/15843.

**Comment by oscardssmith** [comment]
Fixed with 0.7
